library/core/src/intrinsics.rs

```diff
@@ -712,8 +712,14 @@ extern "rust-intrinsic" {
 
     /// Aborts the execution of the process.
     ///
-    /// A more user-friendly and stable version of this operation is
-    /// [`std::process::abort`](../../std/process/fn.abort.html).
+    /// [`std::process::abort`](../../std/process/fn.abort.html) is to be preferred if possible,
+    /// as its behaviour is more user-friendly and more stable.
+    ///
+    /// The current implementation of `intrinsics::abort` (ab)uses a debug trap
+    /// on some popular platforms.
+    /// On Unix, the
+    /// process will probably die of a signal like `SIGABRT`, `SIGILL`, `SIGTRAP`, `SIGSEGV` or
+    /// `SIGBUS`.  The precise behaviour is not guaranteed and not stable.
```

Comment:
Two notes:
- I don't think there's much use in calling this 'abuse'.
- In the rest of `std` we use the word 'terminate' instead of 'die' for processes.

How about something like this?

```suggestion
    /// The current implementation of `intrinsics::abort` results in an invalid
    /// instruction on most platforms. On Unix, this will probably cause the
    /// process to be terminated with a signal like `SIGABRT`, `SIGILL`,
    /// `SIGTRAP`, `SIGSEGV` or `SIGBUS`. The precise behaviour is not
    /// guaranteed and not stable.
```

Reply:
Thaks for the attention.  I have more-or-less adopted your suggestions.  I preferred a more active voice.